solver/simple.go

```diff
@@ -141,6 +140,17 @@ func notifyError(ctx context.Context, st *state, cached bool, err error) {
 	notifyCompleted(err, cached)
 }
 
+func (s *simpleSolver) state(vertex Vertex, job *Job) *state {
+	s.solver.mu.RLock()
+	if st, ok := s.solver.actives[vertex.Digest()]; ok {
+		st.jobs[job] = struct{}{}
+		s.solver.mu.RUnlock()
+		return st
+	}
+	s.solver.mu.RUnlock()
```

Comment:
should this be done after the call to `createState` returns? because `createState` will add the entry into `s.solver.actives`.

otherwise this feels like a race-condition where two threads both call `state` and both end up with a key miss.

Reply:
I think that's correct. There's a lock around the `actives` write on L186, but I think you're right about the race after L145. Will update.